pkg/controller/autoimport/manager.go

```diff
@@ -36,131 +37,105 @@ func Add(ctx context.Context,
 	informerHolder *source.InformerHolder,
 	mcRecorder kevents.EventRecorder) (string, error) {
 
-	c, err := controller.New(controllerName, mgr, controller.Options{
-		Reconciler: NewReconcileAutoImport(
+	err := ctrl.NewControllerManagedBy(mgr).Named(controllerName).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: helpers.GetMaxConcurrentReconciles(),
+		}).
+		WatchesRawSource( // watch the import secrets
+			source.NewImportSecretSource(informerHolder.ImportSecretInformer,
+				&source.ManagedClusterResourceEventHandler{},
+				predicate.Predicate(predicate.Funcs{
+					GenericFunc: func(e event.GenericEvent) bool { return false },
+					DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+					CreateFunc:  func(e event.CreateEvent) bool { return true },
+					UpdateFunc: func(e event.UpdateEvent) bool {
+						new, okNew := e.ObjectNew.(*corev1.Secret)
+						old, okOld := e.ObjectOld.(*corev1.Secret)
+						if okNew && okOld {
+							return !equality.Semantic.DeepEqual(old.Data, new.Data)
+						}
+						return false
+					},
+				}),
+			),
+		).
+		WatchesRawSource( // watch the auto-import secrets
+			source.NewAutoImportSecretSource(informerHolder.AutoImportSecretInformer,
+				&source.ManagedClusterResourceEventHandler{},
+				predicate.Predicate(predicate.Funcs{
+					GenericFunc: func(e event.GenericEvent) bool { return false },
+					DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+					CreateFunc:  func(e event.CreateEvent) bool { return true },
+					UpdateFunc: func(e event.UpdateEvent) bool {
+						new, okNew := e.ObjectNew.(*corev1.Secret)
+						old, okOld := e.ObjectOld.(*corev1.Secret)
+						if okNew && okOld {
+							return !equality.Semantic.DeepEqual(old.Data, new.Data)
+						}
+						return false
+					},
+				}),
+			),
+		).
+		Watches( // watch the managed clusters
```

Comment:
@qiujian16 @zhujian7 

This line of code is the main change to solve auto-import-secret case, for more detaul please see the PR description part.

Others are just log for debugging work in the furture.